lib/SPIRV/libSPIRV/SPIRVModule.cpp

```diff
@@ -348,6 +350,12 @@ class SPIRVModuleImpl : public SPIRVModule {
   std::map<unsigned, SPIRVConstant*> LiteralMap;
 
   void layoutEntry(SPIRVEntry* Entry);
+
+  typedef std::function<bool(SPIRVEntry*, SPIRVEntry*)> IdComp;

```

Comment:
Sorry I did not make it clear. I think DFSState, IdComp, EntryStateMapTy, sortEntries() and visit() should be separated as a helper class. Could be friend class of SPIRVModule and have a pointer to the module as member, or have references to the vectors to be worked on as member.

This will keep SPIRVModule cleaner.